test/tests/bcs/dmg_periodic/tests

```diff
@@ -0,0 +1,58 @@
+[Tests]
+  issues = '#15501'
+  design = 'AddPeriodicBCAction.md  DistributedRectilinearMeshGenerator.md'
+
+  [./check]
+    type = 'RunException'
+    input = 'dmg_periodic_bc.i'
+    cli_args = '/UserObjects/uo/type=CheckGhostedBoundaries /UserObjects/uo/total_num_bdry_sides=160'
+    expect_err = 'All boundaries are ghosted to every single processor, and it is not scalable'
+    requirement = "The system shall be able to determine if a non-scalable algorithm is being used for ghosting boundaries."
+    max_parallel = 1
```

Comment:
This error only works in serial?

Reply:
Yes, I intended to add this test to make sure nobody can delete my code: `CheckGhostedBoundaries` class.

Error is triggered when boundaries are NOT distributed. If there is one core, then boundaries are not distributed in the sense that the local number of side elements equals to the global number of side elements.

Reply:
Boundaries in the mesh generated via `DistributedRectilinearMeshGenerator` are always distributed.  So only way to trigger that little error is to use one core.